lib/nerve.rb

```diff
@@ -86,15 +165,47 @@ def run
       $EXIT = true
     end
 
-    def launch_watcher(name, config)
-      log.debug "nerve: launching service watcher #{name}"
-      watcher = ServiceWatcher.new(config.merge({'instance_id' => @instance_id, 'name' => name}))
-      @watchers[name] = Thread.new{watcher.run}
+    def merged_config(config, name)
+      # Get a deep copy so sub-hashes are properly handled
+      deep_copy = Marshal.load(Marshal.dump(config))
+      return deep_copy.merge({'instance_id' => @instance_id, 'name' => name})
+    end
+
+    def launch_watcher(name, config, wait=false)
+      watcher_config = merged_config(config, name)
+      # The ServiceWatcher may mutate the configs, so record the version before
+      # passing the config to the ServiceWatcher
+      @watcher_versions[name] = watcher_config.hash
+
+      watcher = ServiceWatcher.new(watcher_config)
+      unless @config_manager.options[:check_config]
+        log.debug "nerve: launching service watcher #{name}"
+        watcher_thread = Thread.new { watcher.run }
```

Comment:
That's an interesting idea. We'd have to have a Thread somewhere, so we'll have to deal with concurrent access to instance variables, but ... honestly that's not so bad (since we're just reading them, and they're both simple bools).

I'll poke at this and figure out how much of a refactor that would be.